internal/node/node_repositories.bzl

```diff
@@ -95,297 +163,174 @@ def _prepare_node(repository_ctx):
-
-  Args:
-    repository_ctx: The repository rule context
-  """
-  path = repository_ctx.attr.path
-  for entry in ["bin", "include", "lib", "share"]:
-    repository_ctx.symlink(path + "/" + entry, entry)
-  _prepare_node(repository_ctx)
-
-node_local_runtime = repository_rule(
-  _node_local_runtime_impl,
-  attrs = {
-    "path": attr.string(),
-    "package_json": attr.label_list(),
-    # TODO: change preserve_symlinks default to true all issues with preserve-symlinks resolved
-    "preserve_symlinks": attr.bool(default = False),
-  },
-)
-
-def _prepare_yarn(repository_ctx):
-  """Sets up BUILD files and shell wrappers for the version of Yarn just set up.
-  
-  The BUILD file exports our shell wrapper around Yarn.
-  We wrap yarn in a small JS script that allows us to override PATHs as needed on Windows.
-  We wrap that JS script in a bash script that allows us to override PATHs on Mac/Linux.
-
-  Args:
-    repository_ctx: The repository rule context
-  """
-  node = get_node_label(repository_ctx)
-
-  repository_ctx.file("BUILD.bazel", content="""#Generated by node_repositories.bzl
-package(default_visibility = ["//visibility:public"])
-sh_binary(
-  name = "yarn",
-  srcs = ["yarn.sh"],
-)
-""")
-
-  # Using process.env['PATH'] here to add node to the environment PATH on Windows
-  # before running bin/yarn.js. export PATH="$NODE_PATH":$PATH in yarn.sh below
-  # has no any effect on Windows and setting process.env['PATH'] has no effect
-  # on OSX
-  # TODO: revisit setting node environment PATH when bash dependency is eliminated
-  repository_ctx.file("yarn.js", content="""//Generated by node_repositories.bzl
-const {{spawnSync}} = require('child_process');
-const node = "{}";
-const nodePath = "{}";
-const yarn = "{}";
-process.env['PATH'] = `"${{nodePath}}":${{process.env['PATH']}}`;
-spawnSync(node, [yarn, ...process.argv.slice(2)], {{stdio: ['ignore', process.stdout, process.stderr]}});
+  # This template file is used by the packager tool and the npm_package rule.
+  # `yarn publish` is not ready for use under Bazel, see https://github.com/yarnpkg/yarn/issues/610
+  repository_ctx.file("run_npm.sh.template", content="""
+"{}" "{}" TMPL_args "$@"
 """.format(
-    repository_ctx.path(node),
-    repository_ctx.path(node).dirname,
-    repository_ctx.path("bin/yarn.js")))
+    repository_ctx.path(node_entry),
+    repository_ctx.path(npm_script)))
 
-  repository_ctx.file("yarn.sh", content="""#!/bin/bash
+  # The entry points for yarn for osx/linux and windows
+  # Runs yarn using appropriate node entry point
+  if host.find("windows") == -1:
+    repository_ctx.file("repository_bin/yarn", content="""#!/bin/bash
 #Generated by node_repositories.bzl
+#Executes the given NPM command over each of the package.json folders provided in node_respositories.
 """ + "".join(["""
-NODE="{}"
-NODE_PATH="{}"
-SCRIPT="{}"
-ROOT="{}"
-export PATH="$NODE_PATH":$PATH
-"$NODE" "$SCRIPT" --cwd "$ROOT" "$@"
+"{}" "{}" --cwd "{}" "$@"
```

Comment:
again probably should be named placeholders

Reply:
Good idea